llrt_core/src/modules/http/request.rs

```diff
@@ -207,18 +220,33 @@ fn assign_request<'js>(request: &mut Request<'js>, ctx: Ctx<'js>, obj: &Object<'
                 ));
             }
 
-            request.body = if let Some(blob) = get_class::<Blob>(&body)? {
-                let blob = blob.borrow();
-                Some(TypedArray::<u8>::new(ctx.clone(), blob.get_bytes())?.into_value())
-            } else {
-                Some(body)
+            match get_class::<Blob>(&body)? {
+                Some(blob) => {
+                    let blob = blob.borrow();
+                    request.body =
+                        Some(TypedArray::<u8>::new(ctx.clone(), blob.get_bytes())?.into_value());
+                    request.content_type = match blob.mime_type().len() {
+                        0 => None,
+                        _ => Some(blob.mime_type()),
+                    };
+                },
+                None => {
+                    request.body = Some(body);
+                    request.content_type = None;
+                },
             }
         }
     }
 
     if obj.contains_key("headers").unwrap() {
```

Comment:
If think we can clean up `assign_request` class by avoiding unwraps and use:
```rust
 if let Some(headers) = obj.get_optional("headers"){
        
 }
```
Similarly for "body" and "signal"